src/System.Private.CoreLib/shared/System/IO/TextWriter.cs

```diff
@@ -831,6 +864,13 @@ public override Task WriteAsync(string value)
                 return Task.CompletedTask;
             }
 
+            [MethodImpl(MethodImplOptions.Synchronized)]
+            public override Task WriteAsync(StringBuilder value, CancellationToken cancellationToken = default)
+            {
+                WriteAsync(value, cancellationToken);
```

Comment:
This should be `Write(value);`, otherwise it’s just calling itself over and over.